### ISSUE TITLE
Function ekg-remove-unused-tags not removing tags after enabling embeddings

### ISSUE THREAD

**Issue by jayrajput** [ahyatt/ekg#66]
Function `ekg-removed-unused-tags` or `ekg-clean-db` are not removing unused tags after enabling embeddings for me as the embedding exists for the tag. Here is an example movie tag in my ekg notes without any content.

```
(ekg-tag-used-p "movie")
(:embedding/embedding [-0.020355815 -0.01797225325 -0.0040594423 -0.0263608985 -0.011576385500000001 0.0018720287000000002 -0.011875542749999999 -0.00552889865 -0.00202579035 -0.011079416 0.008641760750000001 0.0342880615 ...])
```
Another tag fire with content shows like this

```
(ekg-tag-used-p "fire")
(:embedding/embedding [-0.016680656 0.00043758826 -0.022189735 0.0046408763 0.0002680446 -0.0012848707 -0.032357123 2.224171e-05 -0.004212005 -0.0044804853 0.011297099 0.011150655 ...] :tag/tagged ("283102e8-b0a8-486f-ab8f-614dee337fd3"))
```

So the embedding shall be deleted when there are no notes for the tag, but I am not sure how to do that.

```
(defun ekg-remove-unused-tags ()
  "Remove all tags that are not used and have no info."
  (cl-loop for tag in (seq-filter (lambda (tag) (not (ekg-tag-used-p tag))) (ekg-tags))
           do
           (ekg-tag-delete tag)))
```

Code which has the problem

**Comment by ahyatt** [comment]
If you run the following, does it fix the problem?
```
(defun ekg-tag-used-p (tag)
  "Return non-nil if TAG has useful information."
  (ekg-connect)
  (triples-get-type ekg-db tag 'tag))
```

**Comment by jayrajput** [comment]
Yes that does, fix the problem

```
(defun ekg-tag-used-p (tag)
  "Return non-nil if TAG has useful information."
  (ekg--connect)
  (triples-get-type ekg-db tag 'tag))
ekg-tag-used-p

(ekg-tag-used-p "movie")
nil


(ekg-tag-used-p "fire")
(:tagged ("283102e8-b0a8-486f-ab8f-614dee337fd3"))
```

**Comment by ahyatt** [comment]
Thank you for testing!  Fixed in 21f82c03e87793766de0e5e49628f972d5decbd7
